File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Model/Expressions/ConstExp.java
package Model.Expressions;

import Model.Utils.MyIDictionary;
import Model.Utils.MyIHeap;

public class ConstExp extends Exp{
    private int number;
    public ConstExp(int number){this.number=number;}
    public int eval(MyIDictionary<String,Integer> tbl, MyIHeap<Integer> hp) {return number;}
    public String toString(){return ""+number;}
}

File: 3rd Semester/Advanced Programming Methods/Laboratory/Toy Interpreter C#/sharplab/Model/Expressions/ConstExp.cs
using sharplab.Model.Utils;
using System;
using System.Collections.Generic;
using System.Text;

namespace sharplab.Model.Expressions
{
    class ConstExp : Exp
    {
        private int number;
        public ConstExp(int number) { this.number = number; }
        public  override int Eval(IMyDictionary<string, int> tbl) { return number; }
        public override string ToString() { return "" + number; }
    }
}
